File: JS/EvilAutocorrect.test.js
/**
 * @version JAVASCRIPT
 * @author MARIO SAM <eu@mariosam.com.br>
 * @see I would love to work with you instead solving web code tests: hire me!
 * $ node NomeDaClasse.test
 */
const TEST = require('tape')
const OBJ = require('./EvilAutocorrect.js')

TEST('Starting EvilAutocorrect test...', (t) => {
    //Test 1
    t.assert(OBJ.autocorrect("We have sent the deliverables to you.") === "We have sent the deliverables to your client.", "Expect: 1 = We have sent the deliverables to your client.")
    //Test 2
    t.assert(OBJ.autocorrect("Our team is excited to finish this with you.") === "Our team is excited to finish this with your client.", "Expect: 2 = Our team is excited to finish this with your client.")
    //Test 3
    t.assert(OBJ.autocorrect("youtube") === "youtube", "Expect: 3 = youtube")
    //Test 4
    t.assert(OBJ.autocorrect("I love youuuuu!") === "I love your client!", "Expect: 4 = I love your client!")
    //Test 5
    t.assert(OBJ.autocorrect("Me and youu forever") === "Me and your client forever", "Expect: 5 = Me and your client forever")
    //Test 6
    t.assert(OBJ.autocorrect("You can do it!") === "your client can do it!", "Expect: 6 = your client can do it!")

    t.end()
})

File: JAVA/EvilAutocorrectTest.java
/** 
 * @version JAVA
 * @author MARIO SAM <eu@mariosam.com.br>
 * @see I would love to work with you instead solving web code tests: hire me!
 * $ mvn clean test -Dtest=your.package.TestClassName
 */ 
package JAVA;

import static org.junit.Assert.assertEquals;
import org.junit.Test;

public class EvilAutocorrectTest {

	@Test
	public void testEvilAutocorrect() throws Exception {
		//Test 1
		String want = "We have sent the deliverables to your client.";
		String got = EvilAutocorrect.autocorrect("We have sent the deliverables to you.");
		assertEquals(want, got);
        //Test 2
        want = "Our team is excited to finish this with your client.";
        got = EvilAutocorrect.autocorrect("Our team is excited to finish this with you.");
		assertEquals(want, got);
        //Test 3
        want = "youtube";
        got = EvilAutocorrect.autocorrect("youtube");
		assertEquals(want, got);
        //Test 4
        want = "I love your client!";
        got = EvilAutocorrect.autocorrect("I love youuuuu!");
		assertEquals(want, got);
        //Test 5
        want = "Me and your client forever";
        got = EvilAutocorrect.autocorrect("Me and youu forever");
		assertEquals(want, got);
        //Test 6
        want = "your client can do it!";
        got = EvilAutocorrect.autocorrect("You can do it!");
		assertEquals(want, got);
	}

}
